libvast/src/system/configuration.cpp

```diff
@@ -47,19 +47,19 @@ void initialize_factories() {
   (factory<Ts>::initialize(), ...);
 }
 
-} // namespace <anonymous>
+} // namespace
 
 configuration::configuration() {
   detail::add_message_types(*this);
   // Use 'vast.conf' instead of generic 'caf-application.ini'.
-  auto config_path_candidates = std::vector{
-    path::current() / "vast.conf",
-  };
-  auto binary = detail::objectpath();
-  if (binary)
-    config_path_candidates.push_back(binary->parent().parent() / "etc" / "vast"
-                                     / "vast.conf");
-  config_path_candidates.emplace_back("/etc/vast/vast.conf");
+  auto config_path_candidates = std::vector<path>{};
+  if (const char* xdg_config_home = std::getenv("XDG_CONFIG_HOME"))
```

Comment:
```suggestion
  if (auto xdg_config_home = std::getenv("XDG_CONFIG_HOME"))
```

Does this work?

Reply:
Nah, we don't want that. The return type of getenv is `char*` for legacy reasons and not `const char*`.

Reply:
But you're not modifying it. It's less overhead to read it with `auto` or even `const auto`.

Reply:
I don't have strong feelings about this, though, so take it or leave it.

Reply:
`const auto` equates to `char* cons`, which is not the same as `const char*`. I prefer to be explicit here.